// @fullcalendar/resource/internal.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var internalCommon = require('./internal-common.js');
require('@fullcalendar/core/internal');
require('@fullcalendar/core/preact');



exports.AbstractResourceDayTableModel = internalCommon.AbstractResourceDayTableModel;
exports.DEFAULT_RESOURCE_ORDER = internalCommon.DEFAULT_RESOURCE_ORDER;
exports.DayResourceTableModel = internalCommon.DayResourceTableModel;
exports.ResourceDayHeader = internalCommon.ResourceDayHeader;
exports.ResourceDayTableModel = internalCommon.ResourceDayTableModel;
exports.ResourceLabelContainer = internalCommon.ResourceLabelContainer;
exports.ResourceSplitter = internalCommon.ResourceSplitter;
exports.VResourceJoiner = internalCommon.VResourceJoiner;
exports.VResourceSplitter = internalCommon.VResourceSplitter;
exports.buildResourceFields = internalCommon.buildResourceFields;
exports.buildRowNodes = internalCommon.buildRowNodes;
exports.flattenResources = internalCommon.flattenResources;
exports.getPublicId = internalCommon.getPublicId;
exports.isGroupsEqual = internalCommon.isGroupsEqual;
exports.refineRenderProps = internalCommon.refineRenderProps;
