// @fullcalendar/resource-timegrid/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@fullcalendar/core');
var premiumCommonPlugin = require('@fullcalendar/premium-common');
var resourcePlugin = require('@fullcalendar/resource');
var timeGridPlugin = require('@fullcalendar/timegrid');
var internalCommon = require('./internal.js');
require('@fullcalendar/core/internal');
require('@fullcalendar/core/preact');
require('@fullcalendar/timegrid/internal');
require('@fullcalendar/resource/internal');
require('@fullcalendar/resource-daygrid/internal');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var premiumCommonPlugin__default = /*#__PURE__*/_interopDefaultLegacy(premiumCommonPlugin);
var resourcePlugin__default = /*#__PURE__*/_interopDefaultLegacy(resourcePlugin);
var timeGridPlugin__default = /*#__PURE__*/_interopDefaultLegacy(timeGridPlugin);

var index = core.createPlugin({
    name: '@fullcalendar/resource-timegrid',
    premiumReleaseDate: '2023-01-11',
    deps: [
        premiumCommonPlugin__default["default"],
        resourcePlugin__default["default"],
        timeGridPlugin__default["default"],
    ],
    initialView: 'resourceTimeGridDay',
    views: {
        resourceTimeGrid: {
            type: 'timeGrid',
            component: internalCommon.ResourceDayTimeColsView,
            needsResourceData: true,
        },
        resourceTimeGridDay: {
            type: 'resourceTimeGrid',
            duration: { days: 1 },
        },
        resourceTimeGridWeek: {
            type: 'resourceTimeGrid',
            duration: { weeks: 1 },
        },
    },
});

exports["default"] = index;
