// @fullcalendar/resource-daygrid/index.esm.js
import { createPlugin } from '@fullcalendar/core';
import premiumCommonPlugin from '@fullcalendar/premium-common';
import resourcePlugin from '@fullcalendar/resource';
import dayGridPlugin from '@fullcalendar/daygrid';
import { ResourceDayTableView } from './internal.esm.js';
import '@fullcalendar/core/internal';
import '@fullcalendar/core/preact';
import '@fullcalendar/daygrid/internal';
import '@fullcalendar/resource/internal';

var index = createPlugin({
    name: '@fullcalendar/resource-daygrid',
    premiumReleaseDate: '2023-01-11',
    deps: [
        premiumCommonPlugin,
        resourcePlugin,
        dayGridPlugin,
    ],
    initialView: 'resourceDayGridDay',
    views: {
        resourceDayGrid: {
            type: 'dayGrid',
            component: ResourceDayTableView,
            needsResourceData: true,
        },
        resourceDayGridDay: {
            type: 'resourceDayGrid',
            duration: { days: 1 },
        },
        resourceDayGridWeek: {
            type: 'resourceDayGrid',
            duration: { weeks: 1 },
        },
        resourceDayGridMonth: {
            type: 'resourceDayGrid',
            duration: { months: 1 },
            // TODO: wish we didn't have to C&P from dayGrid's file
            monthMode: true,
            fixedWeekCount: true,
        },
    },
});

export { index as default };
